Pokedex_ios/View/Controller/SortFilter/SortFilterViewController.swift

```diff
@@ -0,0 +1,132 @@
+           containerView.addSubview(contentStackView)
+           contentStackView.translatesAutoresizingMaskIntoConstraints = false
+           NSLayoutConstraint.activate([
+               contentStackView.topAnchor.constraint(equalTo: containerView.topAnchor, constant: 32),
+               contentStackView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor, constant: -20),
+               contentStackView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor, constant: 20),
+               contentStackView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: -20)
+           ])
+       }
+    func delayDismiss() {
+        _ = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: false) { (_) in
+            self.dismiss(animated: true)
+        }
+    }
+    
+    @IBAction func smallestNumberFirstButton(_ sender: Any) {
+        if let button: UIButton = sender as? UIButton {
+            button.isSelected = !button.isSelected
+            if button.isSelected {
+               button.tintColor = UIColor.white
+                button.backgroundColor = UIColor.applyButtonColor()
+            } else {
+               button.backgroundColor = UIColor.unAppliedButtonColor()
+            }
+        }
+        sortPokemonListViewModel.sortWithSmallestNumberFirst()
+        delayDismiss()
+        }
+
+    @IBAction func highestNumberFirst(_ sender: Any) {
+        if let button: UIButton = sender as? UIButton {
+             button.isSelected = !button.isSelected
+            if button.isSelected {
+               button.tintColor = UIColor.white
+               button.backgroundColor = UIColor.applyButtonColor()
+             } else {
+               button.backgroundColor = UIColor.unAppliedButtonColor()
+             }
+         }
+         sortPokemonListViewModel.sortWithHighestNumberFirst()
+         delayDismiss()
+        }
+
+    @IBAction func aToZ(_ sender: Any) {
+        if let button: UIButton = sender as? UIButton {
+            button.isSelected = !button.isSelected
+           if button.isSelected {
+              button.tintColor = UIColor.white
+              button.backgroundColor = UIColor.applyButtonColor()
+            } else {
+              button.backgroundColor = UIColor.unAppliedButtonColor()
+            }
+         }
+         sortPokemonListViewModel.sortWithAtoZ()
+         delayDismiss()
+        }
+
+     @IBAction func zToA(_ sender: Any) {
+         if let button: UIButton = sender as? UIButton {
+             button.isSelected = !button.isSelected
+            if button.isSelected {
+               button.tintColor = UIColor.white
+            button.backgroundColor = UIColor.applyButtonColor()
+             } else {
+               button.backgroundColor = UIColor.unAppliedButtonColor()
+             }
+         }
+         sortPokemonListViewModel.sortWithZtoA()
+         delayDismiss()
+       }
+}
+
+extension UIColor {
```

Comment:
as this is an UIColor extension, does not belong here, if you´re going to add this, add it to a different class in a folder called extensions or utils

Reply:
True hehe, corrected, thanks :)